### PR TITLE
Resolve macos build issues

### PR DESCRIPTION
**Pull request by twelch**
- [x] Update slonik to 2.0, addressing type inconsistencies
- [x] base-project node_modules symlink follow endless loop
- [ ] issue with jest setup/teardown for scripts test suite

Experimental canary has been published in order to test that bundle features still works in project space - https://www.npmjs.com/package/@seasketch/geoprocessing/v/2.0.1-experimental-sloniky.4

### CODE REVIEW

File: packages/geoprocessing/scripts/dataPrep/bundleFeatures.ts [hunk 1]
```diff
@@ -94,11 +95,13 @@ const bundleFeatures = async (
...
-      npoints: number;
-      bbox: BBox;
-    }>(sql`
+    const recordObject = z.object({
+      id: z.number(),
+      npoints: z.number(),
+      bbox: z.number().array().length(4).or(z.number().array().length(6)),
+    });
+
+    const records = await connection.many(sql.type(recordObject)`
```
Comment (twelch) [inline packages/geoprocessing/scripts/dataPrep/bundleFeatures.ts:104]: slonik is passed the `type` of the query result a different way now.  You pass in a zod object and it should actually now validates the results to match the type at runtime.

File: packages/geoprocessing/scripts/dataPrep/bundleFeatures.ts [hunk 2]
```diff
@@ -121,7 +124,7 @@ const bundleFeatures = async (
     let processedCount = 0;
     for (const feature of records) {
       sumNPoints += feature.npoints;
-      extent = expandBBox(extent, feature.bbox);
+      extent = expandBBox(extent, feature.bbox as BBox);
```
Comment (twelch) [inline packages/geoprocessing/scripts/dataPrep/bundleFeatures.ts:127]: I didn't find a good way to get the zod type to match the BBox type so I punted and did a couple casts.

**Comment by avmey** [comment]
Hm.. this branch isn't completing the npm install either, but isn't complaining about slonik anymore. Attached is the error I get, and the code chunk in npm/prepare.ts it's complaining about. Seems like the cp command isn't building the path correctly on my machine.
Screenshot 2023-05-15 at 2 04 39 PM
Screenshot 2023-05-15 at 2 04 52 PM
Screenshot 2023-05-15 at 2 06 18 PM

**Comment by twelch** [comment]
Looks like maybe an issue with it following a symlink on that recursive copy and it's cycling infinitely.

I think we can exclude something from that copy to make it work, like the node_modules folder in base project path

**Comment by twelch** [comment]
This might be macos specific or zsh specific such that I don't hit it.

**Comment by twelch** [comment]
Decision is we will land this and migrate any remaining info to a new ticket

**Pull request merged**